components/polls/PollForm.tsx

```diff
@@ -144,8 +144,12 @@ const PollForm: React.FC<BoxProps> = ({ ...props }) => {
         block_height: parseInt(isTokenVote ? data.block_height : '0'),
       }]
 
+      const guildId = router.asPath.match(/\d{18,20}/)
+
+
       const clientConfig = [{
         provider_id: "discord",
+        guild_id: guildId![0] ?? '000000000000',
```

Comment:
From an API perspective, can a user submit any guild ID? Or is this validated

Reply:
The api only does basic validation (numberstring)

Reply:
Wouldn't BE have guildid anyway? I'll approve but don't quite understand

Reply:
BE is not talking to discord (as of now). It could in theory take the channel list and query discord to figure out what guild it is. It should become available to BE when we move auth to BE.

Reply:
We're actually not using that filed yet, I thought I'd use it for filtering the cache list. But I think we'll need it eventually.

Reply:
Maybe in the future we use community module, it would ideally contain guildID. Lmk if I should merge

Reply:
Added auth bug fix. We need to merge this simultaneaously with BE deploy.

Reply:
> Maybe in the future we use community module, it would ideally contain guildID. Lmk if I should merge

You're right, the poll should probably have a community ID instead..

Reply:
How should we proceed? We can continue with this and backlog a refactor

Reply:
yes let's go with this for now